cli/catalog-api-v1/src/error.rs

```diff
@@ -3,7 +3,6 @@ use std::hash::Hash;
 use serde::{Deserialize, Serialize};
 
```

Comment:
suggestion: may be a good time to add some additional docs for this here:

```suggestion
/// Type of the error returned by the catalog API
/// Since we were unable to represent earlier error structures returned by the API
/// using the progenitor client generator,
/// errors are now serialized as a blob of values
/// (`context` in [crate::ResolutionMessageGeneral]).
///
/// The context may be parsed into a higher level structure later,
/// or ignored in which case the `message` field in [crate::ResolutionMessageGeneral]
/// is expected to provide a relevant fallback message.
```